File: Arrays_String/LeetCode/Move_Zeroes.cs
using System;
namespace Arrays_string.LeetCode
{
    public class Move_Zeroes
    {
        public void MoveZeroes(int[] nums)
        {

            for (int lastNonZeroFoundAt = 0, cur = 0; cur < nums.Length; cur++)
            {
                if (nums[cur] != 0)
                {
                    Swap(nums, lastNonZeroFoundAt++, cur);
                }
            }
            Console.WriteLine($"[{string.Join(",", nums)}]");

        }

        private void Swap(int[] nums, int p, int q)
        {
            var _copy = nums[p];
            nums[p] = nums[q];
            nums[q] = _copy;
        }
    }
}

File: Arrays_String/LeetCode/Move_Zeroes.go
package Array_String

func moveZeroes(nums []int) {

	i := 0
	for j := 0; j < len(nums); j++ {
		if nums[j] != 0 {
			temp := nums[i]
			nums[i] = nums[j]
			nums[j] = temp
			i++
		}
	}

}
